src/k8s/pkg/utils/sys.go

```diff
@@ -3,20 +3,27 @@ package utils
 import (
 	"context"
 	"fmt"
+	"io"
 	"os"
 	"os/exec"
 )
 
 // runCommand executes a command with a given context.
 // runCommand returns nil if the command completes successfully and the exit code is 0.
 func RunCommand(ctx context.Context, command ...string) error {
+	return RunCommandWithCustomWriter(ctx, os.Stdout, os.Stderr, command...)
+}
+
+// RunCommandWithCustomWriter executes a command with a given context and allows to set a custom writer
+// RunCommandWithCustomWriter returns nil if the command completes successfully and the exit code is 0.
+func RunCommandWithCustomWriter(ctx context.Context, stdout io.Writer, stderr io.Writer, command ...string) error {
```

Comment:
This is somewhat obsolete for this PR but I felt like this can be a useful util for future work. I can remove it if you want.

Reply:
i prefer not to clutter utils, let's only add this in the future if needed